### PR TITLE
Self view refactor

### PR DESCRIPTION
**Pull request by damencho**

### CODE REVIEW

**Comment by saghul** [comment]
Why the refactor?

**Comment by robertpin** [comment]
> Why the refactor?

Some good  feedback here https://community.jitsi.org/t/couple-of-issues-with-hide-self-view/108960

File: config.js [hunk 1]
```diff
@@ -89,6 +89,9 @@ var config = {
     // Disables self-view tile. (hides it from tile view and from filmstrip)
     // disableSelfView: false,
```
Comment (robertpin) [inline config.js:90]: Can you also rename this to hideSelfView so that it's more clear?

File: config.js [hunk 2]
```diff
@@ -89,6 +89,9 @@ var config = {
...
 
+    // Disables self-view settings in UI
+    // disableSelfViewSettings: false,
```
Comment (robertpin) [inline config.js:93]: Add this to configWhitelist

File: react/features/base/settings/functions.any.js [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 import { CONFIG_WHITELIST } from '../config';
+import { getParticipantCount } from '../participants';
 import { toState } from '../redux';
 import { parseURLParams } from '../util';
 
 import { DEFAULT_SERVER_URL } from './constants';
 
 /**
- * Returns the effective value of a configuration/preference/setting by applying
+ * Returns the eff  ective value of a configuration/preference/setting by applying
```
Comment (robertpin) [inline react/features/base/settings/functions.any.js:10]: Unintentional typo

File: react/features/base/settings/functions.any.js [hunk 4]
```diff
@@ -256,3 +257,23 @@ export function shouldHideShareAudioHelper(state: Object): boolean {
...
+}
+
+/**
+ * Gets the disable self view setting.
+ *
+ * @param {Object} state - Redux state.
+ * @returns {boolean}
+ */
+export function getHideSelfView(state: Object) {
+    return state['features/base/config'].disableSelfView || state['features/base/settings'].disableSelfView;
```
Comment (robertpin) [inline react/features/base/settings/functions.any.js:278]: If the config is set to true, there's no way for the user to see the self-view, even if they try to. The user will see the toggle in settings but it won't do anything.
This is why when the config was loaded we updated the settings, so there would be only one option to check.

File: react/features/base/config/middleware.js [hunk 5]
```diff
@@ -125,12 +125,6 @@ function _setConfig({ dispatch, getState }, next, action) {
         }));
     }
 
-    if (action.config.disableSelfView) {
```
Comment (robertpin) [inline react/features/base/config/middleware.js:128]: Removing this causes a problem when we check if we should hide self-view or not. Mentioned in a comment below

**Comment by saghul** [comment]
Gotcha thanks!

File: react/features/notifications/middleware.web.js [hunk 6]
```diff
@@ -1,40 +0,0 @@
...
- * @returns {Function}
- */
-MiddlewareRegistry.register(store => next => action => {
-    switch (action.type) {
-    case CONFERENCE_JOINED: {
-        const { dispatch, getState } = store;
-        const { disableSelfView } = getState()['features/base/settings'];
-
-        if (disableSelfView) {
-            dispatch(showNotification({
```
Comment (robertpin) [inline react/features/notifications/middleware.web.js:27]: I don't think removing this is a good idea. In the case that the config to disableSelfView is set to true (so the self-view is hidden, but the user can show it by going to settings) we have a discoverability issue

**Review by robertpin — CHANGES_REQUESTED** [review]

File: react/features/base/config/middleware.js (hunk 5 above)
Comment (damencho) [reply react/features/base/config/middleware.js:128]: This in general is wrong, you set it to true and then there is no way to set it to false, and if UI is hidden the only way is to edit localstorage.

File: react/features/notifications/middleware.web.js (hunk 6 above)
Comment (damencho) [reply react/features/notifications/middleware.web.js:27]: You see the notification when the self view is not hidden, which is part of the report. And you see it on every join ...

File: react/features/base/settings/functions.any.js (hunk 4 above)
Comment (damencho) [reply react/features/base/settings/functions.any.js:278]: Hum then, maybe when it is controlled from outside we can hide the setting from UI.

Comment (robertpin) [reply react/features/base/settings/functions.any.js:278]: Sounds good to me

File: config.js (hunk 1 above)
Comment (damencho) [reply config.js:90]: We had pushed already as stable with it ... so I do not want to rename it ...

**Comment by damencho** [comment]
@robertpin Can you take a look, also tested and mobile starts and conference works :)

**Review by robertpin — APPROVED** [review]

**Pull request merged**